src/Microsoft.Kusto.ServiceLayer/DataSource/DataSourceBase.cs

```diff
@@ -96,7 +96,8 @@ public async Task<T> ExecuteScalarQueryAsync<T>(string query, CancellationToken
         public abstract string ClusterName { get; }
 
         public abstract string DatabaseName { get; }
-        public abstract GlobalState SchemaState { get; }
+
+        public abstract IIntellisenseClient IntellisenseClient { get; }
```

Comment:
No longer exposing the SchemaState and instead exposing functions that use it within the IntellisenseClient